sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2525,11 +2526,17 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     var properties = ctx.tablePropertyList.asScala.headOption
       .map(visitPropertyKeyValues)
       .getOrElse(Map.empty)
-    Option(ctx.comment).map(string).map {
-      properties += SupportsNamespaces.PROP_COMMENT -> _
+
+    if (properties.keySet.exists(RESERVED_PROPERTIES.contains)) {
+      throw new ParseException(s"Disallow to specify reserved properties, including" +
```

Comment:
we can make the error message better
```
properties.foreach { case (key, _)
  if (RESERVED_PROPERTIES.contains(key)) {
    fail(s"$key is a reserved properties.")
  }
}
```

Reply:
after some more thought, I think this is a breaking change. Existing queries may already use the "comment" property. Can we add a legacy config and update migration guide?

Reply:
OK

Reply:
we have made an undocumented breaking change here, the `comment` and `location` have side-effects now. Simply forbid them seems to make the influence smaller. Otherwise for fully backward compatibility, we shoud add addtional logic to let them have no side effects.
```sql
create database spark2 with dbproperites ('comment'='ct', 'location'='loc');
```
```sql
Database Name	spark2
Description
Location	file:/Users/kentyao/Downloads/spark/spark-2.4.4-bin-hadoop2.7/spark-warehouse/spark2.db
Properties	((comment,ct), (location,loc))
Time taken: 0.022 seconds, Fetched 4 row(s)
```
```sql
create database spark3 with dbproperties ('comment'='ct', 'location'='loc');
```
```sql
Database Name	spark3
Description	ct
Location	file:/Users/kentyao/Downloads/spark/spark-3.0.0-SNAPSHOT-bin-20200103/loc
Owner Name	kentyao
Owner Type	USER
Properties
Time taken: 0.019 seconds, Fetched 6 row(s)
```

Reply:
> we can make the error message better
> 
> ```
> properties.foreach { case (key, _)
>   if (key == PROP_COMMENT) {
>     fail(s"$key is a reserved namespace property, please use the COMMENT clause to specify it.")
>   }
>   ...
> }
> ```

now we have 2 reserved props, after ownership, there shall be 2, what if the number keeps growing :)

Reply:
ah so in Spark 2, the reserved properties are simply ignored?

BTW this error message is for end users, we should try our best to make it helpful, even if it means more code.

Reply:
the reserved properties are not simply ignored but as user-side properties, I guess

Reply:
I think it's OK that the legacy config can only ignore the reserved properties but not treat them as user-side properties. At least the query won't fail.